### ISSUE TITLE
"Unable to connect to X server"

### ISSUE THREAD

**Issue by iDoka** [aivs/blackCockpit#4]
Run `blackCockpit.py` from CLI  (use RPi on text mode):

![Screenshot-from-2019-05-05-16-38-020a62aede6e5d79db.png](http://static2.keep4u.ru/2019/05/05/Screenshot-from-2019-05-05-16-38-020a62aede6e5d79db.png)

How can I fix/suppress warning:  "**Unable to connect to X server**"?

**Comment by aivs** [comment]
Монитор hdmi? Получилось запуститься? В чем проблемы была? 

Для оригинального rpi дисплея, я настраивал только.  Видео сразу запустилось.

Using Official RPi touch display¶

If you are using the official Raspberry Pi touch display, you need to configure Kivy to use it as an input source. To do this, edit the file ~/.kivy/config.ini and go to the [input] section. Add this:

mouse = mouse
mtdev_%(name)s = probesysfs,provider=mtdev
hid_%(name)s = probesysfs,provider=hidinput

**Comment by aivs** [comment]
Нужно выбрать тип дисплея, из документации
https://kivy.org/doc/stable/guide/environment.html#environment

Change the default screen to use¶

You can set an environment variable named KIVY_BCM_DISPMANX_ID in order to change the display used to run Kivy. For example, to force the display to be HDMI, use:

KIVY_BCM_DISPMANX_ID=2 python main.py
Check Controlling the environment to see all the possible values.


KIVY_BCM_DISPMANX_ID
Change the default Raspberry Pi display to use. The list of available value is accessible in vc_dispmanx_types.h. Default value is 0:

0: DISPMANX_ID_MAIN_LCD
1: DISPMANX_ID_AUX_LCD
2: DISPMANX_ID_HDMI
3: DISPMANX_ID_SDTV
4: DISPMANX_ID_FORCE_LCD
5: DISPMANX_ID_FORCE_TV
6: DISPMANX_ID_FORCE_OTHER

**Comment by iDoka** [comment]
да, монитор подключен по HDMI в настольных экспериментах, в перспективе переход на DPI (RGB18).

проблема заключается не в том, что нет картинки, а в этом варнинге - т.е. при запуске даже в ssh приложение открывается на дисплее - это приятно.

При добавлении KIVY_BCM_DISPMANX_ID=2  варнинг ушёл.

В последствии, при переходе на экран DPI непонятно что из этого надо указывать:
```
0: DISPMANX_ID_MAIN_LCD
1: DISPMANX_ID_AUX_LCD
4: DISPMANX_ID_FORCE_LCD
```
